app/assets/javascripts/oxalis/model/bucket_data_handling/pullqueue.js

```diff
@@ -93,56 +93,70 @@ class PullQueue {
     // Measuring the time until response arrives to select appropriate preloading strategy
     const roundTripBeginTime = new Date();
     const layerInfo = getLayerByName(dataset, this.layerName);
+
+    const handleBucket = (bucketAddress, bucketData, resolutions) => {
```

Comment:
As the `pullBatch` method already was pretty long and got even longer now, can you make `handleBucket` an instance method of the pullqueue instead of defining it here, locally?